pytorch_lightning/trainer/connectors/logger_connector/epoch_result_store.py

```diff
@@ -148,48 +148,32 @@ def get_epoch_log_metrics(self, *_, **__) -> List[Dict]:
     def get_forked_metrics(self, *_, **__) -> List[Dict]:
         return self.get_epoch_from_func_name("get_forked_metrics")
 
-    @staticmethod
-    def _append_to_structure(primary_dict, opt_idx, batch_idx, result) -> None:
-        primary_dict.setdefault(opt_idx, {})
-        primary_dict[opt_idx].setdefault(batch_idx, [])
-        primary_dict[opt_idx][batch_idx].append(result)
-
-    def append(self, result, dataloader_idx: Optional[int] = None, extra_info: Optional[dict] = None) -> None:
-        assert isinstance(result, Result)
-        if dataloader_idx is None:
-            dataloader_idx = 0
-
-        if extra_info is None:
-            extra_info = {}
-
-        # [dataloader_idx][optimizer_idx][training_step_idx] is a list
-        if len(extra_info) > 0:
-            self._internal_type = ResultStoreType.INSIDE_BATCH_TRAIN_LOOP
-            # initialize dictionary
+    def append(self, result: Result, info: Dict) -> None:
+        dataloader_idx = info["dataloader_idx"]
+        self._internal_type = info["type"]
+        opt_idx = info["opt_idx"]
```

Comment:
shall it rather uses `.get()` also what if any of the keys is missing?

Reply:
We define the keys. This is not user-facing so if they aren't there either somebody broke it or we have a bug